File: backend/src/main/java/com/ll/nbe342team8/domain/qna/answer/dto/GetResAnswersDto.java
package com.ll.nbe342team8.domain.qna.answer.dto;

import com.ll.nbe342team8.domain.qna.answer.entity.Answer;
import java.util.List;
import java.util.stream.Collectors;

public class GetResAnswersDto {

    List<AnswerDto> answers;

    public GetResAnswersDto(List<Answer> answers) {

        this.answers= answers.stream()
                .map(AnswerDto::new)
                .collect(Collectors.toList());
    }
}

File: backend/src/main/java/com/ll/nbe342team8/domain/qna/answer/dto/GetResAnswersDto.kt
package com.ll.nbe342team8.domain.qna.answer.dto

import com.fasterxml.jackson.annotation.JsonProperty
import com.ll.nbe342team8.domain.qna.answer.entity.Answer

data class GetResAnswersDto(
    @JsonProperty("answers") val answers: List<AnswerDto>
) {
    companion object {
        fun from(answers: List<Answer>): GetResAnswersDto {
            return GetResAnswersDto(answers.map { AnswerDto(it) })
        }
    }
}